pandas/tests/generic/test_frame.py

```diff
@@ -287,3 +289,36 @@ def test_deepcopy_empty(self):
         empty_frame_copy = deepcopy(empty_frame)
 
         self._compare(empty_frame_copy, empty_frame)
+
+    def test_datetimeField_after_setitem(self):
+        # This test covers the unexpected behaviour of datetimeField when using
+        # setitem on another column as reported in issue #6942
+
+        start = pd.to_datetime("20140401")
+
+        df = pd.DataFrame(
+            index=pd.date_range(start, periods=1), columns=["timenow", "Live"]
+        )
+
+        df.at[start, "timenow"] = datetime.today()  # initial time.
+        time1 = df.at[start, "timenow"]
+
+        time.sleep(1)  # sleep time of 1 second in between assignments.
```

Comment:
These `sleep` calls are not needed.

Reply:
Some tests were failing with AssertionError, so added those.

Reply:
Why were `AssertionError`s being raised?

Reply:
I tried the same code on my computer and got the following results:
`time1: 2019-10-04 23:23:31.911241`
`time2: 2019-10-04 23:23:31.911534`
`time3: 2019-10-04 23:23:31.912060`
that is a time difference of 10^(-3) between the time1 and time2.Now if both the statements are executed within a time frame of <= 10^(-6) then the two will match. So that's why I thought of adding the sleep in between time1, time2 and time3 to ascertain this doesn't happen.

Reply:
Sorry I misread the problem and was writing the test to check if the **timenow** column changes **value** after setting the **Live** column with "at".
So, what the test should check is that the  **dtype** of **timenow** column remains unchanged after setting the **Live** column to **True** with **at**. For that I could use the assert_series_equal and check the **dtype** of **timenow** before and after.
Is that right?

Reply:
This test should check that the `timenow` column changes once `df.Live = True`

https://github.com/pandas-dev/pandas/issues/6942#issuecomment-432609912

Reply:
Some points:
1. For the general template of tests that is used in **pandas/tests**, most of them assert that either two DataFrames or Series are equal and call the predefined functions `tm.assert_something_equal`. However in this case we assert the column actually changes value and I couldn't find a function for such a case in `tm` and therefore wrote the individual assert statements for the different values in the `timenow` column(asserting that they are not equal).

2. Without the sleep statements inserted, some checks were failing. When I viewed the log of the failed tests I observed that the error was because
`assert not time1 == time2`
was evaluating to
`assert not datetime.datetime(2019, 10, 4, 16, 12, 38, 392372) == datetime.datetime(2019, 10, 4, 16, 12, 38, 392372)`
So I realized that this could only occur if both the statements are run within a time-frame < 10^(-6). When I added the sleep statements the checks passed.

Reply:
You will still use `tm.assert_frame_equal`. You will have to construct the expected dataframe from scratch

```
In [4]: result = pd.DataFrame(index=pd.date_range(start,periods=1), columns=['timenow','Live'])

In [5]: result.at[start,'timenow'] = datetime.today() # initial value


In [7]: new_date =  datetime.today() 

In [9]: result.Live = True

In [10]: result.at[start,'timenow'] = new_date 

In [11]: expected = pd.DataFrame([[new_date, True]], index=pd.date_range(start,periods=1), columns=['timenow','Live'])

In [12]: tm.assert_frame_equal(result, expected)

```

Reply:
Thanks, couldn't come up with that! Will make the necessary changes and resubmit the PR.